feathr_project/feathr/registry/_feathr_registry_client.py

```diff
@@ -0,0 +1,630 @@
+        return ExpressionTransformation(v['transformExpr'])
+    elif 'defExpr' in v:
+        return WindowAggTransformation(agg_expr=v['defExpr'],
+                                       agg_func=v.get('aggFunc'),
+                                       window=v.get('window'),
+                                       group_by=v.get('groupBy'),
+                                       filter=v.get('filter'),
+                                       limit=v.get('limit'))
+    raise ValueError(f"Invalid transformation format {v}")
+
+
+def feature_type_to_def(v: FeatureType) -> dict:
+    return {
+        "type": v.type,
+        "tensorCategory": v.tensor_category,
+        "dimensionType": [value_type_to_str(t) for t in v.dimension_type],
+        "valType": value_type_to_str(v.val_type),
+    }
+
+
+def dict_to_feature_type(v: dict) -> FeatureType:
+    return FeatureType(val_type=str_to_value_type(v["valType"]),
+                       dimension_type=[str_to_value_type(
+                           s) for s in v["dimensionType"]],
+                       tensor_category=v["tensorCategory"],
+                       type=v["type"])
+
+
+def typed_key_to_def(v: TypedKey) -> dict:
+    ret = {
+        "keyColumn": v.key_column,
+        "keyColumnType": value_type_to_str(v.key_column_type)
+    }
+    if v.full_name:
+        ret["fullName"] = v.full_name
+    if v.description:
+        ret["description"] = v.description
+    if v.key_column_alias:
+        ret["keyColumnAlias"] = v.key_column_alias
+    return ret
+
+
+def dict_to_typed_key(v: dict) -> TypedKey:
+    v = to_camel(v)
+    return TypedKey(key_column=v["keyColumn"],
+                    key_column_type=str_to_value_type(v["keyColumnType"]),
+                    full_name=v.get("fullName"),
+                    description=v.get("description"),
+                    key_column_alias=v.get("keyColumnAlias"))
+
+
+def feature_to_def(v: Feature) -> dict:
+    ret = {
+        "name": v.name,
+        "featureType": feature_type_to_def(v.feature_type),
+        "key": [typed_key_to_def(k) for k in v.key],
+    }
+    if v.transform:
+        ret["transformation"] = transformation_to_def(
+            v.transform)
+    if v.registry_tags:
+        ret["tags"] = v.registry_tags
+    return ret
+
+
+def dict_to_feature(v: dict) -> Feature:
+    ret = Feature(name=v["attributes"]["name"],
+                feature_type=dict_to_feature_type(v["attributes"]["type"]),
+                key=[dict_to_typed_key(k) for k in v["attributes"]["key"]],
+                transform=dict_to_transformation(v["attributes"].get("transformation")),
+                registry_tags=v["attributes"].get("tags", {}))
+    ret._qualified_name = v["attributes"]["qualifiedName"]
+    ret._registry_id = UUID(v["guid"])
+    return ret
+
+
+def _get_type_name(v: Any) -> str:
+    if isinstance(v, Source):
+        return "feathr_source_v1"
+    elif isinstance(v, FeatureAnchor):
+        return "feathr_anchor_v1"
```

Comment:
use `TYPE_DEF_ANCHOR` etc. instead of hard coded string

Reply:
LOL I was looking for this constant in the source code but forgot its exact name, thanks for pointing it out.